lib/utils/registry-secrets.ts

```diff
@@ -0,0 +1,65 @@
+import * as ajv from 'ajv';
+
+export interface RegistrySecrets {
+	[registryAddress: string]: {
+		username: string;
+		password: string;
+	};
+}
+
+/**
+ * JSON schema validator for the private registry secrets (username and
+ * password entries keyed by hostname:port registry addresses).
+ */
+export class RegistrySecretValidator {
+	private registrySecretJsonSchema = {
+		// Sample valid registrySecrets JSON contents:
+		//   {  "docker.example.com": {"username": "ann", "password": "hunter2"},
+		//      "https://idx.docker.io/v1/": {"username": "mck", "password": "cze14"}
+		//   }
+		type: 'object',
+		patternProperties: {
+			'.+': {
+				type: 'object',
+				properties: {
+					username: { type: 'string' },
+					password: { type: 'string' },
+				},
+				additionalProperties: false,
+			},
+		},
+		additionalProperties: false,
+	};
+	private _ajv: ajv.Ajv;
```

Comment:
@CameronDiver @Page- Any idea when we started allowing leading underscores? I thought we were avoiding them for private members and just using them for unused function args (certainly I've always tried to follow this based on the MS TS guidelines).

Reply:
It seems I replied this on the builder PR instead of this one, but yeah we don't use leading underscores.

Reply:
Renamed.